react/OrderShipping.tsx

```diff
@@ -1,31 +1,217 @@
-import React, { createContext, ReactNode, useContext } from 'react'
-
+import React, { createContext, ReactNode, useContext, useCallback } from 'react'
+import { compose, graphql } from 'react-apollo'
+import {
+  QueueStatus,
+  useOrderQueue,
+  useQueueStatus,
+} from 'vtex.order-manager/OrderQueue'
 import { useOrderForm } from 'vtex.order-manager/OrderForm'
 
+import {
+  estimateShipping as EstimateShipping,
+  selectDeliveryOption as SelectDeliveryOption,
+} from 'vtex.checkout-resources/Mutations'
+
 interface Context {
   countries: string[]
+  selectedAddress: CheckoutAddress
+  insertAddress: (address: CheckoutAddress) => void
+  deliveryOptions: DeliveryOption[]
+  selectDeliveryOption: (option: string) => void
 }
 
 interface OrderShippingProps {
   children: ReactNode
+  EstimateShipping: any
+  SelectDeliveryOption: any
```

Comment:
I think those should be in the generics of `graphql` HOC below. Check [this](https://medium.com/@mickeyboston/apollo-typescript-hocs-in-react-explained-93032db3e89a) or some other component that uses Apollo + Typescript

Reply:
Agreed. I just read the medium post and, considering the changes I might have to do, I think it deserves its own PR.